packages/namekit-react/src/components/Identity.tsx

```diff
@@ -0,0 +1,145 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  ReactNode,
+} from "react";
+import {
+  createClient,
+  Network,
+  type SecurePrimaryNameResult,
+} from "@namehash/nameguard";
+
+interface IdentityContextType extends SecurePrimaryNameResult {
+  address: string;
+}
+
+const IdentityContext = createContext<IdentityContextType | null>(null);
+
+const useIdentity = () => {
+  const context = useContext(IdentityContext);
+
+  if (!context) {
+    throw new Error("useIdentity must be used within an IdentityProvider");
+  }
+
+  return context;
+};
+
+interface SubComponentProps {
+  className?: string;
+  children?: ReactNode;
+}
+
+interface RootProps {
+  address: string;
+  network?: Network;
+  className?: string;
+  children: ReactNode;
+}
+
+const Root = ({
+  address,
+  network = "mainnet",
+  className,
+  children,
+  ...props
+}: RootProps) => {
+  const [data, setData] = useState<IdentityContextType | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setLoading(true);
+
+        const nameguard = createClient({ network });
+
+        const result = await nameguard.getSecurePrimaryName(address, {
+          returnNameGuardReport: true,
+        });
+
+        setData({ ...result, address });
+      } catch (err) {
+        setError(
+          err instanceof Error ? err.message : "An unknown error occurred",
+        );
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, [address, network]);
+
+  if (loading) return <div>Loading...</div>;
+  if (error) return <div>Error: {error}</div>;
+  if (!data) return null;
```

Comment:
Suggest a few updates here:

1. The `Root` and all of its children should always be rendered, even if data is still loading or failed to load. The responsibility for handling these cases should be pushed down into child components. Ex: `Avatar` should know what to do when it is loading or when there was a loading error, etc..
2. ... continuing on from above: Suggest updating the interface that is returned by `useIdentity` so that all of the following are true:
   1. Independent of any loading state, it is always guaranteed to be able to get the following values: `address` and `network` (these props should never be undefined).
   2. There's some distinct field(s) representing the loading state / request error state for the identity.
   3. We shouldn't have the returned value `... extends SecurePrimaryNameResult` (because this value might still be loading, etc..). Therefore we should have this as an optional property of the returned value that is undefined unless the loading state field is some value representing it has been successfully loaded.

Reply:
Addressed